### PR TITLE
add roadmap and add contribute guide in readme

### PR DESCRIPTION
**Pull request by fanux**

### CODE REVIEW

**Comment by cuisongliu** [comment]
commit message  add `[skip ci]` try skip ci .

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+* general e2e tests
+* support build, apply and push a CloudImage
+
+## v0.2.0
+
+* support cluster ENV and global config
+* support ARM base image
+* registry cache docker image when pull images
+* base image support more kubernetes versions and auto make rootfs
```
Comment (justadogistaken) [inline ROADMAP.md:13]: auto make rootfs? where to support this?

File: README.md [hunk 2]
```diff
@@ -206,3 +206,8 @@ Some information of the basic settings will be written to the Clusterfile and st
 ```shell script
 sealer delete -f /root/.sealer/my-cluster/Clusterfile
 ```
+
+# Developing Sealer
+
+* [contributing guide](./CONTRIBUTIONG.md)
+* [贡献文档](./docs/contributing_zh.md)
```
Comment (Starnop) [inline README.md:213]: How about also put the CONTRIBUTIONG_ZH.md at the root dir of the project?

File: docs/api/cloudrootfs.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# cloud rootfs
+
+cloud rootfs package all the dependencies when run a kubernetes cluster needs.
```
Comment (justadogistaken) [inline docs/api/cloudrootfs.md:3]: cloud rootfs "will/would" package all the dependencies refers to the kubernetes cluster requirements

**Comment by Starnop** [comment]
> commit message add `[skip ci]` try skip ci .

In fact, the CI will check multiple aspects, including code lint, unit tests, integration tests, markDown lint, shell check, and so on.

I think the CI runner should decide which CI jobs should be run based on the content submitted by the pull request.

File: README.md [hunk 4]
```diff
@@ -80,7 +80,7 @@ Install a kubernetes cluster
 sealer run kubernetes:v1.19.2 --master 192.168.0.2
 ```
 
-If it is installed on the cloud:
+Install a cluster on alicloud:
```
Comment (justadogistaken) [inline README.md:83]: should we mention that we only support alicloud at present, but we are to support other clouds?

File: ROADMAP.md (hunk 1 above)
Comment (fanux) [reply ROADMAP.md:13]: this almost done by @cuisongliu , same as sealos auto make offline package.

**Comment by fanux** [comment]
> > commit message add `[skip ci]` try skip ci .
> 
> In fact, the CI will check multiple aspects, including code lint, unit tests, integration tests, markDown lint, shell check, and so on.
> 
> I think the CI runner should decide which CI jobs should be run based on the content submitted by the pull request.

The golang-ci lint config now support markdown lint?

**Comment by cuisongliu** [comment]
> > commit message add `[skip ci]` try skip ci .
> 
> In fact, the CI will check multiple aspects, including code lint, unit tests, integration tests, markDown lint, shell check, and so on.
> 
> I think the CI runner should decide which CI jobs should be run based on the content submitted by the pull request.

should be so that,may be need add some action image, if not exsits we need redevelop it

File: docs/api/clusterfile.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Clusterfile definition
+
+```yaml
+apiVersion: sealer.aliyun.com/v1alpha1
+kind: Cluster
+metadata:
+  name: my-cluster
+  annotation:
+    sealer.aliyun.com/etcd: "/data/etcd"
```
Comment (cuisongliu) [inline docs/api/clusterfile.md:9]: I think we need tell user support field in clusterfile in different provider.

**Comment by allencloud** [comment]
In terms of ROADMAP of sealer, maybe it is a little bit better to show the status of several phases ambitiously. For developers, roadmap is the direction which lead guys to think big, deep, far and fill guys with passion.  For adopters, roadmap is the direction which makes them have confidence in sealer. For the part of current roadmap content, I am wondering if it is more like a release plan including what we should do in details step by step.

File: README.md (hunk 2 above)
Comment (fanux) [reply README.md:213]: I'am afraid there will be too many file in root dir.

**Comment by fanux** [comment]
> In terms of ROADMAP of sealer, maybe it is a little bit better to show the status of several phases ambitiously. For developers, roadmap is the direction which lead guys to think big, deep, far and fill guys with passion. For adopters, roadmap is the direction which makes them have confidence in sealer. For the part of current roadmap content, I am wondering if it is more like a release plan including what we should do in details step by step.

I will add more details in roapmap, but now for the long-term planning has not been fully agreed, did not think particularly clearly.

File: docs/api/clusterfile.md (hunk 5 above)
Comment (fanux) [reply docs/api/clusterfile.md:9]: A Clusterfile provider field that describes this.

Comment (justadogistaken) [reply docs/api/clusterfile.md:9]: You mean that the concrete provider should provide their specific annotations?  does aliyun provides the annotations already? I think the annotations are little confising, let's add more descriptions?

Comment (fanux) [reply docs/api/clusterfile.md:9]: Some infra information will save in annotations like EIP, the "sealer.aliyun.com/etcd" key is not supported now, just an example

**Review by justadogistaken — APPROVED** [review]
LGTM

**Pull request merged**